cloud/amazon/s3.py

```diff
@@ -296,6 +296,15 @@ def is_walrus(s3_url):
     else:
         return False
 
+def get_md5_digest(local_file):
+    md5 = hashlib.md5()
+    with open(local_file, 'rb') as f:
+        while True:
+            data = f.read(1024 ** 2)
+            if not data: break
```

Comment:
If you do need a while loop and this line is still necessary, break it into two.  Easier to read as two lines than as one.